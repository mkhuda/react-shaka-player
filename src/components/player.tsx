import * as React from "react";
import * as Hooks from "../hooks";
import * as Shaka from "shaka-player/dist/shaka-player.ui";

import { PlayerProps } from "../types/";

const ReactShakaPlayer = (props: PlayerProps) => {
  const videoRef = React.useRef<HTMLVideoElement>(null);
  const uiContainerRef = React.useRef<HTMLDivElement>(null);
  const playerStats = React.useRef<Shaka.extern.Stats | null>(null);

  const { player, ui } = Hooks.usePlayer(videoRef, uiContainerRef, props);
  Hooks.usePlayerListener(player, props);
  Hooks.useUIListener(ui, player, props);
  Hooks.useStats(player, playerStats, props);

  const {
    className,
    playerClassName,
    superConfig,
    config,
    uiConfig,
    onLoad,
    onPlay,
    onPause,
    onEnded,
    onBuffering,
    onStatsChanged,
    onPlayerError,
    ...newProps
  } = props;

  const style = {
    maxWidth: "100%",
    width: "100%",
  };

  return (
    <div
      style={style}
      ref={uiContainerRef}
      className={props.className == undefined ? "mk-theme" : props.className}
    >
      <video
        ref={videoRef}
        className={props.playerClassName}
        style={style}
        {...newProps}
      />
    </div>
  );
};

export { ReactShakaPlayer };
