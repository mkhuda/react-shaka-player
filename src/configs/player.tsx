const streamingConfig = {
  player: {
    manifest: {
      dash: {
        ignoreMinBufferTime: true,
      },
    },
    streaming: {
      lowLatencyMode: true,
      inaccurateManifestTolerance: 0,
      bufferingGoal: 10,
      rebufferingGoal: 1,
    },
  },
  ui: {
    addSeekBar: false,
    controlPanelElements: ["time_and_duration", "mute", "fullscreen"],
    addBigPlayButton: true,
  },
};

const vodConfig = {
  ui: {
    controlPanelElements: ["time_and_duration", "mute", "fullscreen"],
    addBigPlayButton: true,
  },
};

export { streamingConfig, vodConfig };
