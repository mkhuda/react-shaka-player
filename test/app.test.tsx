import * as Enzyme from "enzyme";
import * as React from "react";
import * as renderer from "react-test-renderer";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { ReactShakaPlayer as Player } from "../src/components/player";

Enzyme.configure({ adapter: new Adapter() });

it("Player renders correctly", () => {
  const tree = renderer.create(<Player />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("Player renders source[src] string correctly", () => {
  const props = { src: "this is sample src" };
  const tree = Enzyme.shallow(<Player {...props} />);
  expect(tree).toMatchSnapshot();
});

it("Player render className container correctly", () => {
  const props = { className: "container-class-name" };
  const tree = Enzyme.shallow(<Player {...props} />);
  expect(tree).toMatchSnapshot();
});

it("Player render playerClassName container correctly", () => {
  const props = { playerClassName: "player-class-name" };
  const tree = Enzyme.shallow(<Player {...props} />);
  expect(tree).toMatchSnapshot();
});

it("Player render container className and playerClassName container correctly", () => {
  const props = {
    className: "container-class-name",
    playerClassName: "player-class-name",
  };
  const tree = Enzyme.shallow(<Player {...props} />);
  expect(tree).toMatchSnapshot();
});
