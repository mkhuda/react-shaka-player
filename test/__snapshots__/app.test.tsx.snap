// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player render className container correctly 1`] = `ShallowWrapper {}`;

exports[`Player render container className and playerClassName container correctly 1`] = `ShallowWrapper {}`;

exports[`Player render playerClassName container correctly 1`] = `ShallowWrapper {}`;

exports[`Player renders correctly 1`] = `
<div
  style={
    Object {
      "maxWidth": "100%",
      "width": "100%",
    }
  }
>
  <video
    style={
      Object {
        "maxWidth": "100%",
        "width": "100%",
      }
    }
  />
</div>
`;

exports[`Player renders source[src] string correctly 1`] = `ShallowWrapper {}`;
